UMIP - Add XAUPERL and XAUUSD as supported price identifiers (#93)
UMAprotocol/UMIPs: UMIPs/umip-24.md

## Headers

> | UMIP-24    |                                               |

Reviewer: ```suggestion
| UMIP-26    |                                               |
```

|------------|-----------------------------------------------|

> | UMIP Title | Add XAUPERL as price identifiers              |

Reviewer: ```suggestion
| UMIP Title | Add XAUPERL and XAUUSD as supported price identifiers              |
```

| Authors    | CY (cy@perlin.net)                            |
| Status     | Draft                                         |
| Created    | December 14, 2020                             |

## Summary

> The DVM should support price requests for the XAUPERL price index.

Reviewer: ```suggestion
The DVM should support price requests for the XAUPERL and XAUUSD price indices.
```

## Motivation

> The DVM currently does not support the XAUPERL price index.

Reviewer: ```suggestion
The DVM currently does not support the XAUPERL or XAUUSD price indices.
```

* Pricing of XAUPERL does not exist anywhere yet, but it can be done by dividing XAUUSD price by PERLUSD.

* Pricing of PERLUSD is easily accessible through open centralized exchange APIs. PERL is currently only trading on Binance as outlined in [umip-13](./umip-13.md)

* XAU is referred to gold in troy ounce traded on various exchanges. Pricing of XAUUSD is accessible from data provider’s APIs.

### Opportunity: 

* A synthetic token that tracks the underlying assets would enable better price discovery by making it possible to have cash/PERL settled position on the underlying assets. It could be used as a hedging tool.

## Technical Specification
The definition of this identifier should be:
* Identifier name: XAUPERL
* Base Currency: XAU (troy ounce)
* Quote Currency: PERL
* Source: https://marketdata.tradermade.com
* Result Processing: XAUUSD[Tradermade API] is divided by PERLUSD(T)[Binance].
* Input Processing: None. Human intervention in extreme circumstances where the result differs from broad market consensus.
* Price Steps: 0.00001 (5 decimals in more general trading format)
* Rounding: Closest, 0.5 up
* Pricing Interval: 60 seconds
* Dispute timestamp rounding: down

> 

Reviewer: ```suggestion

The definition of this identifier should be:
* Identifier name: XAUUSD
* Base Currency: XAU (troy ounce)
* Quote Currency: USD
* Source: https://marketdata.tradermade.com
* Result Processing: XAUUSD[Tradermade API].
* Input Processing: None. Human intervention in extreme circumstances where the result differs from broad market consensus.
* Price Steps: 0.00001 (5 decimals in more general trading format)
* Rounding: Closest, 0.5 up
* Pricing Interval: 60 seconds
* Dispute timestamp rounding: down
```

## Implementation

> The value of this identifier for a given timestamp should be determined by querying for the price of PERLUSDT from Binance and XAUUSD from Tradermade’s API for that timestamp.

Reviewer: ```suggestion
The value of the XAUUSD or XAUPERL identifier for a given timestamp should be determined by querying for the price of XAUUSD from Tradermade’s API for that timestamp. To determine the value of XAUPERL, the price of PERLUSDT will also need to be queried from Binance for that timestamp.
```

Users can query “https://marketdata.tradermade.com/api/v1/minute_historical?currency=XAUUSD&date_time=2020-11-11-13:01&api_key=apikey” and use the close price as reference.

> 

Reviewer: ```suggestion
This close price rounded to 5 decimals is the value of XAUUSD.
```

> After querying the price of XAUUSD and PERLUSD at the timestamp, XAUUSD is divided by PERLUSD to get XAUPERL and leave the result as is. This is the settlement price voters should look for.

Reviewer: The results are rounded at this time, not left as is - correct?
